go/vt/vtgate/plan_executor_vschema_ddl_test.go

```diff
@@ -0,0 +1,792 @@
+
+	}
+
+	t.Fatalf("updated vschema did not contain vindexes %v on table %s", names, table)
+	return nil
+}
+
+func TestPlanExecutorAlterVSchemaKeyspace(t *testing.T) {
+	//t.Skip("not yet planned")
+	*vschemaacl.AuthorizedDDLUsers = "%"
+	defer func() {
+		*vschemaacl.AuthorizedDDLUsers = ""
+	}()
+	executor, _, _, _ := createExecutorEnvUsing(planAllTheThings)
+	session := NewSafeSession(&vtgatepb.Session{TargetString: "@master", Autocommit: true})
+
+	vschemaUpdates := make(chan *vschemapb.SrvVSchema, 2)
+	executor.serv.WatchSrvVSchema(context.Background(), "aa", func(vschema *vschemapb.SrvVSchema, err error) {
+		vschemaUpdates <- vschema
+	})
+
+	vschema := <-vschemaUpdates
+	_, ok := vschema.Keyspaces["TestExecutor"].Vindexes["test_vindex"]
+	if ok {
+		t.Fatalf("test_vindex should not exist in original vschema")
+	}
+
+	stmt := "alter vschema create vindex TestExecutor.test_vindex using hash"
+	_, err := executor.Execute(context.Background(), "TestExecute", session, stmt, nil)
+	require.NoError(t, err)
+
+	_, vindex := waitForVindex(t, "TestExecutor", "test_vindex", vschemaUpdates, executor)
+	assert.Equal(t, vindex.Type, "hash")
+}
+
+func TestPlanExecutorCreateVindexDDL(t *testing.T) {
+	*vschemaacl.AuthorizedDDLUsers = "%"
+	defer func() {
+		*vschemaacl.AuthorizedDDLUsers = ""
+	}()
+	executor, sbc1, sbc2, sbclookup := createExecutorEnvUsing(planAllTheThings)
+	ks := "TestExecutor"
+
+	vschemaUpdates := make(chan *vschemapb.SrvVSchema, 4)
+	executor.serv.WatchSrvVSchema(context.Background(), "aa", func(vschema *vschemapb.SrvVSchema, err error) {
+		vschemaUpdates <- vschema
+	})
+
+	vschema := <-vschemaUpdates
+	_, ok := vschema.Keyspaces[ks].Vindexes["test_vindex"]
+	if ok {
+		t.Fatalf("test_vindex should not exist in original vschema")
+	}
+
+	session := NewSafeSession(&vtgatepb.Session{TargetString: ks})
+	stmt := "alter vschema create vindex test_vindex using hash"
+	_, err := executor.Execute(context.Background(), "TestExecute", session, stmt, nil)
+	require.NoError(t, err)
+
+	_, vindex := waitForVindex(t, ks, "test_vindex", vschemaUpdates, executor)
+	if vindex == nil || vindex.Type != "hash" {
+		t.Errorf("updated vschema did not contain test_vindex")
+	}
+
+	_, err = executor.Execute(context.Background(), "TestExecute", session, stmt, nil)
+	wantErr := "vindex test_vindex already exists in keyspace TestExecutor"
+	if err == nil || err.Error() != wantErr {
+		t.Errorf("create duplicate vindex: %v, want %s", err, wantErr)
+	}
+	select {
+	case <-vschemaUpdates:
+		t.Error("vschema should not be updated on error")
+	default:
+	}
+
+	t.Skip("not yet planned")
+
+	// Create a new vschema keyspace implicitly by creating a vindex with a different
+	// target in the session
+	ksNew := "test_new_keyspace"
+	session = NewSafeSession(&vtgatepb.Session{TargetString: ksNew})
+	stmt = "alter vschema create vindex test_vindex2 using hash"
+	_, err = executor.Execute(context.Background(), "TestExecute", session, stmt, nil)
```

Comment:
We do not allow setting target string for non-existing keyspace. So does this test make sense to allow vindex creation on non existing keyspace?